### DISCUSSION TITLE
Primary key over 2 or more fields

### DISCUSSION THREAD

**Discussion by FreeSynergy** [directus/directus#23757] [Feature Requests]
### Summary

A table, for example the postal codes worldwide, are unique in one country, but the postal code exists several times worldwide. If you were to take the country AND the zip code as the primary key, you could use the primary key instead of creating a primary key that is not needed and therefore requires additional security to be built in, which is actually nonsensical.

### Basic Example

An example is:
Zip code and a country_id are defined as primary keys.
However, several other fields can also be defined as primary keys.

### Motivation

This feature can make several fields unique together. This not only saves a lot of work, but also a lot of speed. It is an elementary feature that can actually be used at the lowest database level. I would like to use this to make things faster and easier.

### Detailed Design

Instead of the primary key being created initially, each field can be defined as a primary key (as a Boolean value). If the field is activated, the fields are defined as primary keys in the database. If they are deselected, the field is deselected again as primary key in the database (i.e. all fields defined as primary key are defined as new primary key, even if there are fewer or no fields).

### Must Have:
- Bool-field, that makes every Field as a Primary Key.

### Drawbacks

Maybe you think, this is more easy, without this feature. But in my opinion it isn't, because if I do a mistake in the first step, I can not delete the primary key, nor something else. I have to delete the hole Table, to change the primary key. This makes no sense …

### Alternatives

I don't know an alternative … because this is so elementary …

### Adoption Strategy

You don't need to write a migration. You only have to store the nower fields, that are primary keys, then you have to add only one more data: The primary key value. That's it …

### Unresolved Questions

In my case … no …

**Comment by FreeSynergy** [comment]
Really … nobody cares about a elementary function?

**Comment by hanneskuettner** [comment]
This is a duplicate feature request to the composite key feature that has gathered a lot of upvoted already. See https://github.com/directus/directus/discussions/12137

I'll proceed to close this, if the original request is missing anything you mentioned, please add a comment, but please refrain from commenting "+1" and instead use the upvote feature.
